### PR TITLE
Add twisted versions of the brainpool curves to common code and testcases

### PR DESCRIPTION
**Pull request by ifranzki**
EP11 also supports the twisted versions of the brainpool curves.

### CODE REVIEW

File: usr/lib/pkcs11/common/mech_ec.c [hunk 1]
```diff
@@ -31,31 +31,61 @@ CK_BYTE brainpoolP160r1[] = {
     0x02, 0x08, 0x01, 0x01, 0x01
 };
 
+CK_BYTE brainpoolP160t1[] = {
```
[the remark this answers is not in the document]
Comment (ifranzki) [reply usr/lib/pkcs11/common/mech_ec.c]: Good point... All the others were without const, so I added the new once without const, too.
Do you want me to change all of them to const while we are at it?

File: testcases/crypto/ec_func.c [hunk 2]
```diff
@@ -193,18 +210,24 @@ static unsigned int curve_len(int index)
...
-    case 3: return CURVE256_LENGTH/8;
-    case 4: return CURVE320_LENGTH/8;
-    case 5: return CURVE384_LENGTH/8;
-    case 6: return CURVE512_LENGTH/8;
-    case 7: return CURVE512_LENGTH/8;
-    case 8: return CURVE192_LENGTH/8;
-    case 9: return CURVE224_LENGTH/8;
-    case 10: return CURVE256_LENGTH/8;
+    case 1: return CURVE160_LENGTH/8;
+    case 2: return CURVE192_LENGTH/8;
```
Comment (p-steuer) [inline testcases/crypto/ec_func.c]: fall-through for equal return values

File: usr/lib/pkcs11/common/mech_ec.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (p-steuer) [reply usr/lib/pkcs11/common/mech_ec.c]: you can give it a try. if it produces a lot of warnings requiring a lot of rework to fix, better not do it.

File: testcases/crypto/ec_func.c [hunk 3]
```diff
@@ -90,22 +88,35 @@ CK_ULONG total_assertions = 65;
...
     CK_ULONG size;
+    CK_BBOOL twisted;
 } _ec_struct;
 
 /* Supported Elliptic Curves */
-#define NUMEC		13
+#define NUMEC		19
 CK_BYTE brainpoolP160r1[] =
     { 0x06, 0x09, 0x2B, 0x24, 0x03, 0x03, 0x02, 0x08, 0x01, 0x01, 0x01 };
+CK_BYTE brainpoolP160t1[] =
```
Comment (p-steuer) [inline testcases/crypto/ec_func.c]: const

File: usr/lib/pkcs11/common/mech_ec.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ifranzki) [reply usr/lib/pkcs11/common/mech_ec.c]: Hmm, well, I get this on all '&brainpool….' when I try to make them const:
../common/mech_ec.c:117:58: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-array-qualifiers]
     {BRAINPOOL_CURVE, CURVE320, sizeof(brainpoolP320t1), &brainpoolP320t1},

File: testcases/crypto/ec_func.c (hunk 3 above)
Comment (ifranzki) [reply testcases/crypto/ec_func.c]: Same here:
ec_func.c:145:6: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-array-qualifiers]
     {&brainpoolP512r1, sizeof(brainpoolP512r1), CK_FALSE},
      ^

File: usr/lib/pkcs11/common/mech_ec.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (p-steuer) [reply usr/lib/pkcs11/common/mech_ec.c]: so leave it like it is for now. we ll look for qualifier thing later

[the remark this answers is not in the document]
Comment (ifranzki) [reply usr/lib/pkcs11/common/mech_ec.c]: fixed

File: testcases/crypto/ec_func.c (hunk 2 above)
Comment (ifranzki) [reply testcases/crypto/ec_func.c]: fixed

File: testcases/crypto/ec_func.c (hunk 3 above)
Comment (ifranzki) [reply testcases/crypto/ec_func.c]: fixed

**Pull request merged**